### DISCUSSION TITLE
How to set user agent header in tweepy client

### DISCUSSION THREAD

**Discussion by priscilla-kroll** [tweepy/tweepy#1825] [Questions] [answered]
Hi, I keep getting RemoteDisconnected-ProtocolError-ConnectionError issue while scraping tweets, so I thought I'd try setting a header as suggested [here](https://stackoverflow.com/questions/33174804/python-requests-getting-connection-aborted-badstatusline-error) and [here](https://stackoverflow.com/questions/48105448/python-http-server-client-remote-end-closed-connection-without-response-error).

The old tweepy seemed to have this [enabled](https://groups.google.com/g/tweepy/c/kx49SsWY8TA?pli=1) for Cursor and Stream. How do I do this with the current tweepy.Client v2?

**Comment by Harmon758** [comment, marked as answer]
`Client` has a [`user_agent` attribute](https://tweepy.readthedocs.io/en/latest/client.html#tweepy.Client.user_agent) you can set.

**Reply by priscilla-kroll** [reply]
That seems like an attribute I can access rather than something I can set, is that right?
How do I set it like in the suggestion links above?
![image](https://user-images.githubusercontent.com/96188877/155829675-bc57ac02-be6b-4924-ac2e-6d6231d66f84.png)

**Reply by Harmon758** [reply]
You can set it prior to making requests.
It's an attribute, not an argument you can pass on initialization.

**Reply by priscilla-kroll** [reply]
Do you mind showing me an example code of how to set it please? I'm confused, so is it an attribute that I can just "check" afterwards, or is this an argument that I can set within tweepy.Client()'s parameters?

**Reply by Harmon758** [reply]
You can set the attribute after initialization, prior to making requests, but it isn't an argument you can pass on initialization:
```py
client = tweepy.Client("Bearer Token here")
client.user_agent = "User Agent here"
...
```

**Reply by priscilla-kroll** [reply]
Ah understood, thanks so much!
On a related note, do you know how I can try this [end_header](https://stackoverflow.com/questions/48105448/python-http-server-client-remote-end-closed-connection-without-response-error) method with tweepy?

**Reply by Harmon758** [reply]
I'm not sure what you're asking. That question involves using an HTTP server (with [`http.server.BaseHTTPRequestHandler`](https://docs.python.org/3/library/http.server.html#http.server.BaseHTTPRequestHandler)), which has nothing to do with Tweepy.

**Reply by priscilla-kroll** [reply]
Hmm ok, I'll have to understand this deeper. Thanks a lot for your help!
